### ISSUE TITLE
Access levels in fileed

### ISSUE THREAD

**Issue by b18emigu** [HGustavs/LenaSYS#8397]
At the moment it seems really unclear how this is meant to be working, me and @a18felen was working on issue #7897 and noticed how the only way to access the page is by being a **superuser**.

What we are uncertain about is if anyone else than **superusers** should even be able to access the page at all? And if that is the case, why dont **superusers** have write access on all files no matter what? 

It seems odd that you cant even access the page if youre a **studentteacher** for the course with `R` rights, as an example. It would seem to us that **studentteachers** should also be able to access this page but only see the course local files. (no global files) Currently the user **brom** is one of the only users that can access the page but he does not have write access for the course which doesnt seem to make sense. 

The way we see it and have tried to understand it is as following:

**Superuser**: Full CRUD access for files in any course and global files
**Student teacher**: Read access for files in the course? Maybe even edit? Upload?
**write access**: Course specific teachers that have CRUD access for files in a specific course?

TL:DR - **superusers** can have acccess file handling but does not necessarily have `write access` meaning they cant delete or upload files. Who should be able to access `fileed.php` and what should they be able to see based on their access level?

**Comment by a97marbr** [comment]
**Superuser** should always have access to everything and be able to do anything.
**Write access** is course based meaning that they can add/delete/update for a specific course (including hidden files and files that require that your logged in)
**Read access** can read files that are public and files that require that you are logged in and has course access 
**Guest** can read public files

**Comment by a18felen** [comment]
@a97marbr Does that mean that people with write access should have access to the file editor? Because at the moment only people with superuser has access.

Or should even people with read access have access to fileed? but not being able to add/delete/update?

**Comment by a97marbr** [comment]
Well,

The fileEd is where we add/update/delete files so I would say that only users with those priviliges should be able to use fileEd. Superuser and write access users. 

We must make sure that write access users only can add/update/delete in those courses where they have write access.

Viewing files are done from showdoc.php and showdugga.php

**Comment by a18felen** [comment]
What rights should student teachers have then? Because there was code in fileEd that stated that student teachers should not have access to upload files but its code that suggest that they have access to be in fileEd.

**Comment by a97marbr** [comment]
I would propose a new access level ... not sure if `studentteacher` is appropriate though since it does not necessary need to be a student that is helping out in a course.

Perhaps **supervisor**? Can see and perhaps modify existing content (?) but not add content.

**Comment by b18emigu** [comment]
So would the **studentteacher** role be removed and replaced with **supervisor**, or would both exist?
